### ISSUE TITLE
Number of Image tokens in input_ids mismatched with num_images when finetuning LLaVA-NeXT

### ISSUE THREAD

**Issue by lihytotoro** [huggingface/transformers#34047]
### System Info

Copy-and-paste the text below in your GitHub issue and FILL OUT the two last points.

- `transformers` version: 4.45.1
- Platform: Linux-5.4.0-135-generic-x86_64-with-glibc2.31
- Python version: 3.10.8
- Huggingface_hub version: 0.25.1
- Safetensors version: 0.4.5
- Accelerate version: 1.0.0
- Accelerate config:    not found
- PyTorch version (GPU?): 2.4.1+cu121 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: <fill in>
- Using GPU in script?: <fill in>
- GPU type: NVIDIA A100-SXM4-80GB

### Who can help?

@zucchini-nlp 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

I encountered this bug when trying to finetune llava-v1.6-vicuna-7b-hf with the repository [https://github.com/zjysteven/lmms-finetune](url).
1. Clone and Installation
git clone https://github.com/zjysteven/lmms-finetune.git
conda create -n lmms-finetune python=3.10 -y
conda activate lmms-finetune
python -m pip install -r requirements.txt
python -m pip install --no-cache-dir --no-build-isolation flash-attn
2. Running example_image.sh
`bash example_scripts/example_image.sh`

The warning I got seems to indicate that I need to merge the image tokens into input_ids earlier, but I cannot figure out where to do this(in the dataset or collator?)
![warning](https://github.com/user-attachments/assets/6ab1263d-e2ff-42ba-a2a9-3c37cf1d7209)
Then in the trainer I got this error:
![error](https://github.com/user-attachments/assets/4301e053-3669-4020-9c24-e1160a70c8a6)
Which probably means the image tokens in input_ids are not recognized properly?

### Expected behavior

Expect behaviour: the image tokens are merged into input_ids successfully in the Trainer, and thus the finetuning proceeds normally.

**Comment by zucchini-nlp** [comment]
Hey @lihytotoro !

The reproduction is quite lengthy and involves a lot of code in the "lmms-finetune" repo. Can you provide a small reproducer that causes the same error.

From traceback I guess the reason is that padding sides do not match. Make sure that `model.padding_side == processor.tokenizer.padding_side`. For that I'd recommend to use a prev version of transformers as we had a small update on tokenizers and I don't see the fix on latest patch release. So, can you try to downgrade transformers first and then follow the above steps?

cc @ArthurZucker , can we get https://github.com/huggingface/transformers/pull/33942 in the next patch release, if there are any. If next minor release is soon, don't worry, we can wait for a while :)

**Comment by ArthurZucker** [comment]
Next patch release is soon yep!

**Comment by zucchini-nlp** [comment]
closing as the fix was merged
